dominion/dominion_ui/src/components/Search/index.js

```diff
@@ -32,37 +32,44 @@ class Search extends React.Component {
       geography: [],
       results: []
     };
-
     this.handleSearch = this.handleSearch.bind(this);
+    this.loadSuggestions = this.loadSuggestions.bind(this);
   }
 
-  async componentDidMount() {
-    const api = createAPI();
+  componentDidMount() {
     let geography = [];
-    if (window.selected_country) {
-      geography = await api.getGeography(5656);
-    } else {
-      geography = Object.values(window.dominion_countries).map(country => ({
-        name: country.name,
-        type: 'country'
-      }));
+    const countries = window.dominion_countries;
+    geography = Object.keys(window.dominion_countries).map(slug => ({
+      slug,
+      name: countries[slug].name,
+      type: 'country'
+    }));
+    this.setState({ geography });
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { searchTerm } = this.state;
+    if (prevState.searchTerm !== searchTerm) {
+      this.loadSuggestions(searchTerm);
     }
+  }
 
-    this.setState({
-      geography
-    });
+  async loadSuggestions(searchTerm) {
+    const api = createAPI();
+    let { geography } = this.state;
+    let countryCode;
+    if (window.selected_country && searchTerm !== '') {
+      countryCode = window.selected_country.code;
+      geography = await api.getGeography(countryCode, searchTerm);
```

Comment:
Do we need to filter these again ? [here](https://github.com/CodeForAfrica/HURUmap-apps/pull/153/files#diff-a8725bcb80038cc1e9e5021ab43e209eR65)

Reply:
We will need it if we are not in country